version: '3'

volumes:
  jupyter:
    driver: local

networks:
  traefik_network:
    external: true

services:
  jupyter:
    image: jupyter/datascience-notebook
    networks:
      - traefik_network
    labels:
      - "traefik.enable=true"
      - "traefik.docker.network=traefik_network"
      - "traefik.port=8888"
      - "traefik.entryPoint=https"
      - "traefik.backend=jupyter"
      - "traefik.frontend.rule=Host:jupyter.mysite.com"
    volumes:
     - jupyter:/home/jovyan/work
    environment:
     - JUPYTER_ENABLE_LAB=yes
    command: start-notebook.sh --NotebookApp.token='changethis' --no-browser --allow-root --notebook-dir=/home/jovyan/work
